src/inferencerequest.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <string>
+#include <unordered_map>
+
+#include "inferenceparameter.hpp"
+#include "inferencetensor.hpp"
+#include "pocapi.hpp"
+
+namespace ovms {
+
+class Status;
+
+class InferenceRequest {
+    const std::string servableName;
+    const uint64_t servableVersion;
+    std::unordered_map<std::string, InferenceParameter> parameters;
+    std::unordered_map<std::string, InferenceTensor> inputs;
+
+public:
+    InferenceRequest(const char* modelName, uint64_t modelVersion);
```

Comment:
What is the reason of using const char* in our cpp implementation? Why not to use string here?

Reply:
This will be directly called by C-API. With async API it makes sense to have ownership of servable name kept in request & it doest not make sens to create temp std::string just to pass this into InferenceRequest constructor I think.